CONTRIBUTING.md

```diff
@@ -4,6 +4,19 @@ Thanks for helping make KEDA better 😍.
 
 There are many areas we can use contributions - ranging from code, documentation, feature proposals, issue triage, samples, and content creation.
 
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
```

Comment:
That's great, but how does this work? Should add some docs on this :)

Reply:
I would say that's enough to google `doctoc` to find out more information. Also not sure where to put such docs

Reply:
Yeah. And wit pre-commits you basically do not have to worry about it - just install pre-commit and you forget about it..

Reply:
That's a bit of an odd argument as that would mean we don't have to doc the pre-commit thingy but it's ok.

> Yeah. And wit pre-commits you basically do not have to worry about it - just install pre-commit and you forget about it..

We tend to make it easy for folks to contribute so you cannot assume everybody has `pre-commit` installed, that's why we have a CI.

Reply:
> We tend to make it easy for folks to contribute so you cannot assume everybody has pre-commit installed, that's why we have a CI.

As we have totally lightweight pre-commit setup we should probably encourage people to use them. It's good to teach people how to automate stuff and focus on the most important things like the code itself and documentation. However, that's only my personal opinion.

Reply:
That's definitely the case and I agree, but we can't expect everyone to use it.

Reply:
We also have it integrated on the CI, so if someone does not use pre-commit locally, they will get a readable diff on the CI that will show what changes should be made.

Reply:
Yes, that was my point on https://github.com/kedacore/keda/pull/1025#discussion_r476294569 indeed

Reply:
Yeah. Pre-commits are working like a charm on Apache Airflow for ? year now and we have plenty of pre-commits running/ You can see it working here for example: https://github.com/apache/airflow/runs/1026196376?check_suite_focus=true#step:7:108 - it integrates very well with CI. you simply run `pre-commit run --all-files` in the CI and you get the very same checks executed in CI as locally. 

And as @mik-laj mentioned - with `--show-diff-on-failure` flag it will even show you what modification would be made if you run pre-commit locally. In case it fails in CI it will even tell you what to do 'pip install pre-commit' and then `pre-commit run <id of pre-commit that failed>`.  But you still can fix it manually if you prefer (though a lot of pre-commit checks can fix things for you automatically (like end-of-file checking, sorting import in python etc.) and then the only thing you have to do you have to commit modified files. And you have literally hundreds of hooks that you can start using. 

So you do not have to even install it as pre-commit (though it is super-easy `pre-commit install` will do the job after installit it with pip. The nice thing about it is that you simply forget about it once you install - it will automatically run all the precommits that were added by others since and you will actually fix the problems even before others see them.

I do not see any drawbacks of using pre-commit to be honest  if your project's goal is to keep some consistency, it's the perfect tool for the job IMHO.

Reply:
Don't get me wrong, I'm not saying it's a bad thing or so; I was just saying that it's fine and the CI will cover it but we shouldn't force people to use it.

I'm not saying you are, it was just a general remark :) This escalated quickly.